src/dependencies.py

```diff
@@ -406,7 +446,14 @@ async def get_api_key_user(
                 detail="API key not associated with a valid user",
                 headers={"WWW-Authenticate": "Bearer"},
             )
-            
+        
+        # ========================================================================
+        # CACHE WRITE-BACK (Best effort, non-blocking)
+        # ========================================================================
+        # Cache the validated API key → user_id mapping for future requests
+        # This is fire-and-forget - if it fails, we still return success
+        await cache_user_id(key_prefix, user.id)
```

Comment:
It's not true "fire-and-forget" as it waits for the result of execution before moving to "return user". So it delays user from getting a response (we wait for save/fail). If it's expected - good. 
If not - we need to change each save (and maybe invalidate) execution to this:

> asyncio.create_task(cache_user_id(key_prefix, user.id))